### PR TITLE
Nest result state

### LINKED ISSUE #338: `baseLanguage`, `languages` reducers not covered

### PR DESCRIPTION
**Pull request by ycleptkellan**
- Nests result state.
- Reworks our thunk tests a bit, as these changes require us to pass a more complete initial state object to our `createMockStore` helper.
- Resolves #338.

### CODE REVIEW

**Comment by ycleptkellan** [comment]
@ericnewcomer this one's ready

**Comment by ericnewcomer** [comment]
screen shot 2018-05-31 at 3 04 07 pm

Seeing two keys for suggestedNameCount in dev tools

**Comment by ycleptkellan** [comment]
Thanks that should be gone now.

File: src/component/NodeEditor/NodeEditor.tsx [hunk 1]
```diff
@@ -1281,7 +1281,7 @@ const mapStateToProps = ({
     typeConfig,
     resultName,
     showResultName,
-    suggestedResultNameCount,
+    suggestedResultNameCount: suggestedNameCount,
```
Comment (ericnewcomer) [inline src/component/NodeEditor/NodeEditor.tsx]: seems no reason these can't have the same name

**Pull request merged**